integration-tests/hibernate-orm-tenancy/connection-resolver/src/main/java/io/quarkus/it/hibernate/multitenancy/AgroalRuntimeDataSourceCreationFeature.java

```diff
@@ -0,0 +1,13 @@
+package io.quarkus.it.hibernate.multitenancy;
+
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+import io.agroal.pool.ConnectionHandler;
+
+public class AgroalRuntimeDataSourceCreationFeature implements Feature {
```

Comment:
Is it normal that this is only done in tests? Isn't it a problem for anyone doing something similar?

Reply:
People certainly will have the same problem if they do something similar. But we don't really recommend creating Agroal datasources at runtime, and we don't test it much either. People probably can create connections through some other ways, too.

I can move this to the Agroal extension, if you want? (using build items, of course)